### PR TITLE
refactor: move apidocs to tessellation

### PR DESCRIPTION
**Pull request by gclaramunt**
Add API docs and publishing action from apidocs project

### CODE REVIEW

File: .github/workflows/release-apidocs.yml [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  pull_request:
+
+name: Publish API docs
+
+env:
+  GITHUB_TOKEN: ${{ github.token }}
+  AWS_REGION:
+    "us-west-1"
+
+  #        if: github.ref == 'refs/heads/master' && github.event_name == 'push'
```
Comment (marcinwadon) [inline .github/workflows/release-apidocs.yml:17]: leftover?

File: .github/workflows/release-apidocs.yml [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+      - integrationnet
+      - develop
+  pull_request:
+
+name: Publish API docs
+
+env:
+  GITHUB_TOKEN: ${{ github.token }}
+  AWS_REGION:
+    "us-west-1"
```
Comment (marcinwadon) [inline .github/workflows/release-apidocs.yml:15]: probably should be taken from env/secret

File: .github/workflows/release-apidocs.yml [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+  #        if: github.ref == 'refs/heads/master' && github.event_name == 'push'
+jobs:
+  terraform:
+    name: Terraform
+    environment:
+      name: ${{ (github.ref == 'refs/heads/mainnet' && 'mainnet') || (github.ref == 'refs/heads/integrationnnet' && 'integrationnet') || (github.ref == 'refs/heads/testnet' && 'testnet') || 'dev' }}
```
Comment (marcinwadon) [inline .github/workflows/release-apidocs.yml:22]: we do have different branches here. `release/mainnet`, `release/integrationnet` and `release/testnet`

File: .github/workflows/release-apidocs.yml [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        id: init
+        run: terraform init
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+
+      - name: Terraform workspace select
+        working-directory: apidocs/terraform
+        id: workspace
+        run: terraform workspace select ${{ (github.ref == 'refs/heads/mainnet' && 'mainnet') || (github.ref == 'refs/heads/integrationnnet' && 'integrationnet') || (github.ref == 'refs/heads/testnet' && 'testnet') || 'dev' }}
```
Comment (marcinwadon) [inline .github/workflows/release-apidocs.yml:69]: aa

**Review by marcinwadon — COMMENTED** [review]
We already have a `doc/` directory with `adr/` inside. Maybe we should use `doc/api/` for the API documentation? It would be well organized then

File: apidocs/terraform/modules/openapi/main.tf [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+      module_path = path.module
+      spec_path = var.spec_path
+      s3_bucket_name = var.s3_bucket_name
+      s3_prefix = replace(var.s3_prefix, "/", "\\/")
+      s3_file_name = var.s3_default_file
+    })
+  }
+}
+
+// aws s3 rm s3://apidoc-dev.constellationnetwork.io --recursive
```
Comment (marcinwadon) [inline apidocs/terraform/modules/openapi/main.tf:34]: leftover?

Comment (gclaramunt) [reply apidocs/terraform/modules/openapi/main.tf:34]: heh, yeah, was exactly like that in the original

File: .github/workflows/release-apidocs.yml [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+            })
+
+      - name: Terraform Plan Status
+        working-directory: apidocs/terraform
+        if: steps.plan.outcome == 'failure'
+        run: exit 1
+
+      - name: Terraform Apply
+        working-directory: apidocs/terraform
+        if: (github.ref == 'refs/heads/mainnet' || github.ref == 'refs/heads/develop' || github.ref == 'refs/heads/integrationnet' || github.ref == 'refs/heads/testnet') && github.event_name == 'push'
```
Comment (kpudlik) [inline .github/workflows/release-apidocs.yml:136]: as @marcinwadon  mentioned above

File: .github/workflows/release-apidocs.yml (hunk 2 above)
Comment (gclaramunt) [reply .github/workflows/release-apidocs.yml]: will keep it hardcoded for now

**Comment by marcinwadon** [comment]
@gclaramunt hey, is it ready for review or you're still testing it?

**Review by marcinwadon — APPROVED** [review]

**Pull request merged**